quarkchain/core.py

```diff
@@ -177,12 +177,18 @@ def deserialize(self, bb):
 
 
 class PrependedSizeMapSerializer:
+    """
+    does not serialize/deserialize biguint values that evalulate to False
+    """
+
     def __init__(self, size_bytes, key_ser, value_ser):
         self.size_bytes = size_bytes
         self.key_ser = key_ser
         self.value_ser = value_ser
 
     def serialize(self, item_map: Dict, barray):
+        if self.value_ser is biguint:
```

Comment:
I am not a big fan of this - PrependedSizeMapSerializer should be generally purpose map serializer which should produce exactly the same map after deserialization.  I would suggest adding an argument in ctor like skip_func(k, v), and TokenBalanceMap should provide lambda k, v: v == 0 as the argument.

Reply:
sounds good, will do that, I don't like this either but didn't thought of a better way